lib/Onyx.js

```diff
@@ -410,6 +482,26 @@ function keyChanged(key, data, canUpdateSubscriber) {
         if (subscriber.withOnyxInstance) {
             // Check if we are subscribing to a collection key and overwrite the collection member key value in state
             if (isCollectionKey(subscriber.key)) {
+                // If the subscriber has a selector, then the consumer of this data must only be given the data
+                // returned by the selector and only when the selected data has changed.
+                if (subscriber.selector) {
+                    subscriber.withOnyxInstance.setState((prevState) => {
+                        const prevData = prevState[subscriber.statePropertyName];
+                        const newValue = getSubsetOfData(data, subscriber.selector);
+                        const newData = {
+                            [key]: newValue,
+                        };
```

Comment:
This is a collection key subscriber so we want to include the previous collection data + the new item that changed. By setting newData to an object of just the new key/value we will overwrite the collection.

Reply:
Maybe we can do an automated test for this if one doesn't exist.

Reply:
There is a test for this case, but it wasn't probably fully fledged out to cover this edge. I've improved it like you suggested (I think).